src/validators/passport.js

```diff
@@ -7,81 +7,131 @@ const reBook = '^[a-zA-Z0-9]{9}$'
+ *
+ */
+const validateHasPassportBranch = (hasPassports) => {
+  if (!hasPassports) {
+    return false
   }
 
-  validHasPassports() {
-    if (!this.hasPassports) {
-      return false
-    }
+  if (!(hasPassports === 'No' || hasPassports === 'Yes')) {
+    return false
+  }
 
-    if (!(this.hasPassports === 'No' || this.hasPassports === 'Yes')) {
-      return false
-    }
+  return true
+}
 
+/**
+ * This validates if the hasPassport branch was answered "yes"
+ *
+ */
+export const validateHasPassport = (hasPassports) => {
+  if (hasPassports === 'Yes') {
     return true
   }
+  return false
+}
 
-  validName() {
-    if (this.hasPassports === 'No') {
-      return true
-    }
 
-    return new NameValidator(this.name).isValid()
+const validateName = (hasPassports, name) => {
+  if (hasPassports === 'No') {
+    return true
   }
 
-  validPassportNumber() {
-    if (this.hasPassports === 'No') {
-      return true
-    }
+  return new NameValidator(name).isValid()
+}
 
-    if (!this.number || !this.number.value) {
-      return false
-    }
+const validatePassportNumber = (hasPassports, passportNumber, issuedDate) => {
+  if (hasPassports === 'No') {
+    return true
+  }
 
-    if (this.issued) {
-      const cutoffDate = new Date('1/1/1990 00:00')
-      const issueDate = extractDate(this.issued)
+  if (!passportNumber || !passportNumber.value) {
+    return false
+  }
 
-      let re = new RegExp(reBook)
+  if (issuedDate) {
+    const cutoffDate = new Date('1/1/1990 00:00')
+    const issueDate = extractDate(issuedDate)
 
-      // Before 1/1/1990 allow alphanumeric of any length
-      if (issueDate && issueDate < cutoffDate) {
-        re = new RegExp('^[a-zA-Z0-9]*$')
-        if (!re.test(this.number.value)) {
-          return false
-        }
-      }
+    let re = new RegExp(reBook)
 
-      // After 1/1/1990 enforce default regex
-      if (!re.test(this.number.value)) {
+    // Before 1/1/1990 allow alphanumeric of any length
+    if (issueDate && issueDate < cutoffDate) {
+      re = new RegExp('^[a-zA-Z0-9]*$')
+      if (!re.test(passportNumber.value)) {
         return false
       }
     }
 
+    // After 1/1/1990 enforce default regex
+    if (!re.test(passportNumber.value)) {
+      return false
+    }
+  }
+  return true
+}
+
+const validateDates = (hasPassports, issuedDate, expirationDate) => {
+  if (hasPassports === 'No') {
     return true
   }
 
-  validDates() {
-    if (this.hasPassports === 'No') {
-      return true
-    }
+  const range = {
+    from: issuedDate,
+    to: expirationDate,
+    present: false,
+  }
+  return new DateRangeValidator(range).isValid()
+}
 
-    const range = {
-      from: this.issued,
-      to: this.expiration,
-      present: false,
-    }
-    return new DateRangeValidator(range).isValid()
+export const validatePassport = (data = {}) => {
+  const {
+    Name, Number, Issued, Expiration, HasPassports,
+  } = data
+
+  const hasPassports = HasPassports.value
+
+  return (
```

Comment:
since these functions all have the same logic:
```
if (hasPassports === 'No') {
  return true
}
```

I think it would be cleaner to check for that inside this fn and return early instead of executing all 4 functions